pandas/core/indexes/api.py

```diff
@@ -66,7 +66,11 @@
 
 
 def get_objs_combined_axis(
-    objs, intersect: bool = False, axis: Axis = 0, sort: bool = True, copy: bool = False
+    objs,
+    intersect: bool = False,
+    axis: Axis = 0,
+    sort: bool | np.bool_ = True,
```

Comment:
Maybe instead wrap arguments to this function in a bool, i.e. `get_objs_combined_axis(..., sort=bool(sort), ...)`?

If not that, maybe in in  the `_typing` have union types, e.g. `any_bool = Union[bool, np.bool_]`, `any_int = union[int, np.int_]` etc.?